docs/features/pouch_with_upgrade.md

```diff
@@ -0,0 +1,102 @@
+# Upgrade Container
+
+## Introduction
+
+Most containers within Alibaba are used in rich container mode. Some containers among the rich containers in traditional virtual machine O&M mode are still stateful. Updates and upgrades of stateful services are frequently performed daily operations within Alibaba. For the container technology that delivers images, the container operations corresponding to service update and upgrade include two steps: deletion of old image container and creation of the new image container.
+
+The stateful service upgrade requires that the new container must inherit all resources of old container, such as network and storage. The following two cases are the service provision scenarios of rich containers:
+
+* Customer Case 1: In a database service, remote data is downloaded to the local computer as the initial data in the database when the container service is created for the first time. Database initialization lasts for a long time, so the new container must inherit the data stored on the old container in the consequent service upgrades to shorten the service provision time.
+
+* Customer Case 2: In a middleware service, the service registration mode is used. That is, the IP addresses of the scaled-up containers must be registered in the server list; otherwise, the containers are invalid. The new containers must inherit the IP addresses of old containers every time the service containers are upgraded; otherwise, the new services are invalid.
+
+## Design
+
+Before using the `upgrade` interface, we should figure out which parameter of container you want to update ? If you want to update a container's [Resources](../../apis/types/resources.go), I think you should use the `update` interface. If you just want to update a container's image and inherit the volumes and network from the old container, now you should use the `upgrade` interface.
+
+### About API
+
+Now, `upgrade` api request body is below, we only support you to specify the Image which used to create a new container and the new `Entrypoint` and `Cmd` that the new container used to run.
+
+```golang
+type ContainerUpgradeConfig struct {
+
+	// Execution commands and args
+	// Min Items: 1
+	Cmd []string `json:"Cmd"`
+
+	// The entrypoint for the container as a string or an array of strings.
+	// If the array consists of exactly one empty string (`[""]`) then the entry point is reset to system default.
+	//
+	Entrypoint []string `json:"Entrypoint"`
+
+	// image
+	// Required: true
+	Image string `json:"Image"`
+}
+```
+
+It is enough to update a container from a old image to a new one now. If anyone has new situation need to add more parameters into `ContainerUpgradeConfig`, we can create a new issue to discuss it.
```

Comment:
s/a old/an old 

When you are using `upgrade` feature, please try to use word `upgrade` rather than `update`.
I think this would reduce something ambiguous.

Reply:
you are right